python/ray/train/lightning/lightning_trainer.py

```diff
@@ -0,0 +1,108 @@
+from typing import TYPE_CHECKING, Type, Optional, Dict
+
+import pytorch_lightning
+
+from ray.util import PublicAPI
+from ray.train.torch import TorchTrainer
+
+if TYPE_CHECKING:
+    from ray.train.torch import TorchConfig
+    from ray.air.config import ScalingConfig, DatasetConfig, RunConfig
+    from ray.train.trainer import GenDataset
+    from ray.data.preprocessor import Preprocessor
+    from ray.air.checkpoint import Checkpoint
+
+
+@PublicAPI(stability="alpha")
+class LightningTrainer(TorchTrainer):
+    """A Trainer for data parallel PyTorch Lightning training.
+
+    This Trainer runs the ``pytorch_lightning.Trainer.fit()`` method on multiple
+    Ray Actors. The training is carried out in a distributed fashion through PyTorch
+    DDP. These actors already have the necessary torch process group already
+    configured for distributed PyTorch training.
+
+    The training function ran on every Actor will first initialize an instance
+    of the provided ``lightning_module`` class object, which is a subclass of
+    ``pytorch_lightning.LightningModule`` using the arguments provided in
+    ``lightning_module_init_config``. The training function will then convert the
+    Ray Dataset shards to a ``pytorch_lightning.LightningDataModule``. Then, the
+    training function will initialize an instance of ``pytorch_lightning.Trainer``
+    using the arguments provided in ``trainer_init_config`` and then run
+    ``pytorch_lightning.Trainer.fit``.
+
+    If the ``datasets`` dict contains a training dataset (denoted by the "train"
+    key), then it will be split into multiple dataset shards, with each Actor
+    training on a single shard. All the other datasets will not be split. To specify
+    a batch size for your datasets, you should set the ``batch_size`` key of
+    ``lightning_module_init_config``. Note that in your ``LightningModule``'s
+    ``__init__``, you should not accept a ``batch_size`` argument.
```

Comment:
Is it possible to use a different batch size for training and validation datasets with this API?

Maybe we can have a `dataloader_config: Dict[str, Dict[str, Any]]` to map the dataset name to args that should be passed in when creating the DataLoader.

Reply:
I use `pytorch_lightning.LightningDataModule.from_datasets`, which accepts a single batch-size argument for all dataloaders. We can re-implement this function ourselves but allow different batch sizes for each dataset, which would allow us to use a `dataloader_config`

Reply:
Ok got it that sounds good.

Reply:
Should we just have batch size as a separate arg? Might make it more clear than having as part of the unit config.